chain/indexer.go

```diff
@@ -444,14 +444,25 @@ func (t *TipSetIndexer) Close() error {
 			log.Errorw("error received while closing task processor", "error", err, "task", name)
 		}
 	}
+	for name, p := range t.messageProcessors {
+		if err := p.Close(); err != nil {
+			log.Errorw("error received while closing message task processor", "error", err, "task", name)
+		}
+	}
 	for name, p := range t.actorProcessors {
 		if err := p.Close(); err != nil {
 			log.Errorw("error received while closing actor task processor", "error", err, "task", name)
 		}
 	}
+
+	return nil
+}
+
+func (t *TipSetIndexer) Close() error {
 	// ensure there are no persist go routines left running
 	t.persistSlot <- struct{}{}
```

Comment:
Im wondering if:
```go
if len(t.persistSlot) == 0 {
  return
} else {
  t.persistSlot <- struct{}{}
}
```
might be safer?

Reply:
That's potentially racy. Another way is to include a timeout:

```Go
select {
    case t.persistSlot <- struct{}{}:

    case time.After(3*time.Second):

}
```

Reply:
I think what's here is fine for now. I don't have enough inside into the time it takes to persist models to make an accurate guess on an appropriate timeout.